src/core/collision.cpp

```diff
@@ -598,7 +598,7 @@ void handle_collisions() {
       // ore one is ghost and one is not accessible
       // we only increase the counter for the ext id to use based on the
       // number of particles created by other nodes
-      if ((!p1 or p1->l.ghost) and (!p2 or p2->l.ghost)) {
+      if ((!p1 or p1->l.ghost) and (!p2 or p2->l.ghost) or !p1 or !p2) {
```

Comment:
I think the two `!px` in the `and` are now superfluous. This should be equivalent to:
```c++
if (!p1 || !p2 || (p1->l.ghost && p2->l.ghost))
```